### PR TITLE
allow text-transformation like uppercase and lowercase

### PR DESCRIPTION
**Pull request by itinance**
This Pull-request allows an **optional** new style-attribute called "textTransformation" to upper-case or lower-case the text of a tag like in CSS the attribute "text-transform" does.  

As in CSS, valid values are "lowercase" and "uppercase":

Example-Code of an stylesheet for htmlviewer:

```
 h: {
    ...baseStyle,
    fontFamily: 'SFUIText-Bold',
    textTransformation: 'uppercase',
  },

```

### CODE REVIEW

**Comment by richchurcher** [comment]
@itinance Seems like a reasonable idea. Can you update to remove the conflict, and please submit only the change you want to make without alterations to other coding style? So `==` as that's the way it's currently written for string comparison, and use of `let/const` rather than `var`. Much appreciated!

File: htmlToElement.js [hunk 1]
```diff
@@ -22,28 +22,38 @@ function htmlToElement(rawHtml, opts, done) {
         if (rendered || rendered === null) return rendered
       }
 
-      if (node.type == 'text') {
+      if (node.type === 'text') {
+        var data = node.data;
```
Comment (richchurcher) [inline htmlToElement.js]: Please use `let/const` throughout.

File: htmlToElement.js [hunk 2]
```diff
@@ -22,28 +22,38 @@ function htmlToElement(rawHtml, opts, done) {
...
-          <Text key={index} style={parent ? opts.styles[parent.name] : null}>
-            {entities.decodeHTML(node.data)}
+            <Text key={index} style={parent ? opts.styles[parent.name] : null}>
+              {entities.decodeHTML(data)}
           </Text>
         )
       }
 
-      if (node.type == 'tag') {
+      if (node.type === 'tag') {
```
Comment (richchurcher) [inline htmlToElement.js]: I appreciate `===` is in more common use, but since this is string comparison let's stick with the project's existing style. If you feel strongly this should be changed, submit a different PR :)

File: htmlToElement.js [hunk 3]
```diff
@@ -22,28 +22,38 @@ function htmlToElement(rawHtml, opts, done) {
...
+            break;
+          case 'lowercase':
+            data = data.toLowerCase();
+            break;
+        }
         return (
-          <Text key={index} style={parent ? opts.styles[parent.name] : null}>
-            {entities.decodeHTML(node.data)}
+            <Text key={index} style={parent ? opts.styles[parent.name] : null}>
+              {entities.decodeHTML(data)}
```
Comment (richchurcher) [inline htmlToElement.js]: Please retain existing indent style. Thanks!

**Comment by richchurcher** [comment]
This is fairly stale now, closing but happy to accept a PR on a more recent version if you'd still like to have this feature included!

**Comment by RobertSheaO** [comment]
Please do! I was just looking for this ability today. Also, can we call it "textTransform" instead of "textTransformation"? The latter isn't an exact match for the actual CSS property.
